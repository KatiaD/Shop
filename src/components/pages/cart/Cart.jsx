import React from 'react';
import { Link } from 'react-router-dom';

import { Master } from 'composables';
import { Button, Input } from 'components/controls';
import { CartItem } from 'components/widgets';
import { CartTable } from './Cart.styled';

const displayName = 'Category';

function Cart({
  myProducts,
  total,
  user,
  handlePlusItem,
  handleMinusItem,
  handleSubmit,
  handleSetUser,

}) {

  return (
    <Master title="Cart">
      {total ? (
        <CartTable>
          <thead>
            <tr>
              <th>Product name</th>
              <th>Price</th>
              <th>Quantity</th>
              <th>total</th>ÃŸ
            </tr>
          </thead>
          <tbody>
            {myProducts.map(product => (
              <CartItem key={product.id} {...product} plus={handlePlusItem} minus={handleMinusItem} controls="true" />
            ))}
          </tbody>
        </CartTable>
      ) : (
          'Cart is empty'
        )}
      {total ? `TOTAL: ${total}` : <Link to="/">Back to catalog</Link>}
      {
        total ?
          <div>
            <Input name="name" placeholder="Type your name" value={user.name ? user.name : ''} onChange={handleSetUser} />
            <Input name="address" placeholder="Type your address" value={user.address ? user.address : ''} onChange={handleSetUser} />
            <Button onClick={handleSubmit}>Submit</Button>
          </div> : ''
      }
    </Master>
  );
}

Cart.displayName = displayName;

export default Cart;