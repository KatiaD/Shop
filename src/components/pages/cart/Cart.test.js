import React from 'react';
import { shallow } from 'enzyme';

import Cart from './Cart';

const testProps = {
  myProducts: [],
  total: 500,
  handlePlusItem: jest.fn(),
  handleMinusItem: jest.fn(),
  handlerSubmit: jest.fn(),
  user: {},
  quantity: {},
};

describe('Given a  CartPage component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeAll(() => {
      component = shallow(<Cart {...testProps} />);
    });

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
