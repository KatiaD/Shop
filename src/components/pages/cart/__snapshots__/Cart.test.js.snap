// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a  CartPage component when the component is rendered should match the snapshot 1`] = `
<Master
  title="Cart"
  total={500}
>
  <Cartstyled__CartTable>
    <thead>
      <tr>
        <th>
          Image
        </th>
        <th>
          Product name
        </th>
        <th>
          Price
        </th>
        <th>
          Quantity
        </th>
        <th>
          total
        </th>
      </tr>
    </thead>
    <tbody />
  </Cartstyled__CartTable>
  TOTAL: 500
  <div>
    <ReduxForm />
  </div>
</Master>
`;
